src/Resources/app/storefront/src/checkout/confirm-order.plugin.js

```diff
@@ -280,12 +284,55 @@ export default class ConfirmOrderPlugin extends Plugin {
             } else {
                 this.mountPaymentButton(paymentMethodInstance);
             }
-
         } catch (e) {
             console.log(e);
         }
     }
 
+    renderPrePaymentButton(componentConfig, selectedPaymentMethodObject) {
+        const PRE_PAY_BUTTON = Object.assign(componentConfig.extra, selectedPaymentMethodObject, {
+            configuration: selectedPaymentMethodObject.configuration,
+            amount: {
+                value: adyenCheckoutOptions.amount,
+                currency: adyenCheckoutOptions.currency,
+            },
+            onClick: (resolve, reject) => {
+                if (!componentConfig.onClick(resolve, reject, this)) {
+                    return false;
+                }
+                ElementLoadingIndicatorUtil.create(document.body);
+            }
+        });
+        let paymentMethodInstance = this.adyenCheckout.create(
+            selectedPaymentMethodObject.type,
+            PRE_PAY_BUTTON
+        );
+        this.mountPaymentButton(paymentMethodInstance);
+    }
+
+    completePendingPayment(paymentMethodType, config) {
+        const url = new URL(location.href);
+        // Check for pending payment session
+        if (url.searchParams.has('pending')) {
```

Comment:
Can we use 'pending' as a config constant like config.sessionKey?

Reply:
the original idea was to use `pending` as a generic constant across all components, but actually since I'm passing the component config into this method, I think I can get rid of `pending` and simply check for the `config.sessionKey` instead

Reply:
That's better yes